### PR TITLE
Set task started_at on INITIALIZING to RUNNING phase transition

### PR DESCRIPTION
**Pull request by hamersaw**
# TL;DR
Currently the "started_at" field is set on task execution on phase transition from QUEUED or UNDEFINED to RUNNING. FlytePropeller may sometimes send an event for the INITIALIZING phase, which is reported from the k8s watch API indicating the pod is in fact initializing. This PR enables setting the started_at field on transitions from INITIALIZING to RUNNING. 

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/1945

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by hamersaw** [comment]
@EngHabu @katrogan Is there any reason to not set the the started_at field on every transition from a non RUNNING phase to the RUNNING phase? It could simplify this conditional.

**Review by EngHabu — APPROVED** [review]
